unified-codes/apps/data-service/src/app/resolvers.ts

```diff
@@ -62,10 +39,16 @@ export const resolvers = {
         console.log(`Entity requested by anonymous user.`);
       }
 
-      const { type = 'medicinal_product' } = _args?.filter ?? {};
+      const { type = 'medicinal_product' } = filter ?? {};
       const query = queries.entities(type);
       const response = await dgraph.postQuery(query);
-      return response.data.query;
+      const allEntities: Array<IEntity> = response.data.query;
+      const paginationParameters: IPaginationParameters<IEntity> = {
+        first,
+        offset,
+        results: allEntities,
+      };
```

Comment:
having the class is nice. agree would be good to update this.

Reply:
Since `EntityCollection` doesn't say anything about the semantics of what it contains (i.e. could be all entities, could be a subset with a particular predicate, etc. etc.), I think it will also be good design for refactoring later when we move to back-end pagination. Sound right, or nah?